### ISSUE TITLE
confluence connector: recursive indexing only uses ancestors

### ISSUE THREAD

**Issue by nisi99** [onyx-dot-app/onyx#3312]
If a page is currently indexed via the `page_id` with all ancestor (i.e. `index_recursively = True`) **only** the ancestors are used.
```
        elif page_id:
            # if a cql_query is not provided, we will use the page_id to fetch the page
            if index_recursively:
                cql_page_query += f" and ancestor='{page_id}'"
            else:
                cql_page_query += f" and id='{page_id}'"
```

**Possible Fix** - simply replace `and` with `or`:
```
        elif page_id:
            # if a cql_query is not provided, we will use the page_id to fetch the page
            cql_page_query += f" and id='{page_id}'"
            if index_recursively:
                cql_page_query += f"or ancestor='{page_id}'"
```

**Comment by mboret** [comment]
I've also found it counter-intuitive to index only the ancestors, not the page himself. 
Furthermore, if you want to index only one page (so you disable the 'index recursively'), it will fail (0 document indexed). 

From my point of view, by default, when you define a page ID, with or without the 'index_recursively' option, the page should be indexed.

A workaround to tackle this is to define your proper CQL request in the connector configuration for example "type=page and id=123456" to index a single page or "type=page and (id=123456 or ancestor=123456)" to index the page and the ancestors.

**Comment by emerzon** [comment]
Fixed in #3503

**Comment by hagen-danswer** [comment]
good catch! fix should be in now :)
